.github/workflows/pr-checks.yaml

```diff
@@ -2,103 +2,49 @@ name: Pull request checks
 
 on: [pull_request]
 
-jobs:
-  build-environment:
-    runs-on: ubuntu-18.04
-    name: Build conda environment
-    steps:
-      - uses: actions/checkout@v2
-        name: Checkout repository
-
-      # See: https://github.com/marketplace/actions/setup-conda
-      - name: Setup anaconda
-        uses: s-weigand/setup-conda@v1
-        with:
-          conda-channels: "conda-forge"
-
-      # Build cache of environment
-      - name: Cache conda environment
-        id: cache-environment
-        uses: actions/cache@v2
-        with:
-          path: nxviz.tar.gz
-          # Note: Remember that whatever files the environment build depends on
-          # should be hashed and added to the key.
-          key: ${{ runner.os }}-env.${{ hashFiles('environment.yml') }}
-
-      - name: Build environment
-        if: steps.cache-environment.outputs.cache-hit != 'true'
-        run: |
-          conda env create -f environment.yml
-          python -m pip install .
-
-      - name: Install conda-pack
-        if: steps.cache-environment.outputs.cache-hit != 'true'
-        run: conda install -c conda-forge conda-pack
-
-      - name: Run conda-pack
-        if: steps.cache-environment.outputs.cache-hit != 'true'
-        run: conda pack -n nxviz -o nxviz.tar.gz
-
-      # See: https://github.com/actions/upload-artifact
-      - name: Upload environment
-        uses: actions/upload-artifact@v2
-        with:
-          name: nxviz-tarball
-          path: nxviz.tar.gz
+env:
+  cache-version: "cache-v2"
 
-  unit-tests:
-    name: Run unit tests
+jobs:
+  pre-commit:
     runs-on: ubuntu-latest
-    needs: build-environment
-
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - uses: pre-commit/action@v3.0.1
```

Comment:
I think this section can be moved to `code-style.yaml` and replace the existing set of jobs. If we standardize on pre-commit hooks doing all code style checking, there will be no risk of conflict between CI and local.

Reply:
i think this isn't functioning as a hook actually; it just fails and kills the action if i didn't fix the formatting properly. 

do you need to login to pre-commit.ci to make it work again?

Reply:
Oh, I see -- that's actually the desired behavior of pre-commit when running on CI/CD. Locally, it should fail if any hooks fail (and those hooks will fail if say style checks are not adhered to), but if we committed skips past the pre-commit hooks and still pushes, then running pre-commit hooks on CI/CD Will ensure that those issues are still surfaced up.

Reply:
So I don't think we need to login to pre-commit.ca to make it work. For now, if they are failing on CI/CD, we can merge while still letting the checks fail, and perhaps defer fixes till later.